documentation/src/test/java/org/hibernate/search/documentation/search/query/QueryDslIT.java

```diff
@@ -378,6 +382,78 @@ public void fetchSize() {
 		} );
 	}
 
+	@Test
+	public void json_elasticsearch() {
+		Assume.assumeTrue( backendConfiguration instanceof ElasticsearchBackendConfiguration );
+
+		OrmUtils.withinJPATransaction( entityManagerFactory, entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+			// tag::elasticsearch-requestTransformer[]
+			List<Book> hits = searchSession.search( Book.class )
+					.extension( ElasticsearchExtension.get() ) // <1>
+					.predicate( f -> f.match()
+							.field( "title" )
+							.matching( "robot" ) )
+					.requestTransformer( context -> { // <2>
+						Map<String, String> parameters = context.getParametersMap(); // <3>
+						parameters.put( "search_type", "dfs_query_then_fetch" );
+
+						JsonObject body = context.getBody(); // <4>
+						body.addProperty( "min_score", 0.5f );
+					} )
+					.fetchHits( 20 ); // <5>
+			// end::elasticsearch-requestTransformer[]
+
+			assertThat( hits ).extracting( Book::getId )
+					.containsExactlyInAnyOrder( BOOK1_ID, BOOK3_ID );
+		} );
+
+		OrmUtils.withinJPATransaction( entityManagerFactory, entityManager -> {
+			SearchSession searchSession = Search.session( entityManager );
+			// tag::elasticsearch-responseBody[]
+			ElasticsearchSearchResult<Book> result = searchSession.search( Book.class )
+					.extension( ElasticsearchExtension.get() ) // <1>
+					.predicate( f -> f.match()
+							.field( "title" )
+							.matching( "robt" ) )
+					.requestTransformer( context -> { // <2>
+						JsonObject body = context.getBody();
+						body.add( "suggest", jsonObject( suggest -> { // <3>
+							suggest.add( "my-suggest", jsonObject( mySuggest -> {
+								mySuggest.addProperty( "text", "robt" );
+								mySuggest.add( "term", jsonObject( term -> {
+									term.addProperty( "field", "title" );
+								} ) );
+							} ) );
+						} ) );
+					} )
```

Comment:
IMHO that seems too hard to read, and maybe to write as well. Why do not we write the JSON a string instead? Maybe we can do it even if here we're merging objects...

Reply:
We could write the JSON as a string in this specific cases, but users will, in most cases, want to generate the JSON dynamically, and then concatenating strings opens the door to injection vulnerabilities. Which is why I didn't want to give an example that would give the impression that using strings is a good idea.

And yes, I agree it's still not very readable even with the utility method. It'll have to do; it's an advanced feature, after all.